foundation-extensions/experience-platform-extension/edge-network-api-reference.md

```diff
@@ -278,7 +350,325 @@ public class EdgeEventHandle {
+    *
+    * @param xdm {@code Map<String, Object>} of raw XDM schema data
+    * @param datasetIdentifier The Experience Platform dataset identifier where this event is sent.
+    *                          If not provided, the default dataset defined in the configuration ID is used
+    * @return instance of current builder
+    * @throws UnsupportedOperationException if this instance was already built
+    */
+  public Builder setXdmSchema(final Map<String, Object> xdm, final String datasetIdentifier) {...}
+
+  /**
+    * Solution specific XDM event data for this event, passed as raw mapping of keys and
+    * Object values.
+    * If XDM schema is set multiple times using either this API or {@link #setXdmSchema(Schema)},
+    * the value will be overwritten and only the last changes are applied.
+    * Setting {@code xdm} to null clears the value.
+    *
+    * When using {@link Edge#sendEvent} this event is sent to the default
+    * Experience Platform dataset defined when generating the Experience Edge configuration ID.
+    *
+    * @param xdm {@code Map<String, Object>} of raw XDM schema data
+    * @return instance of current builder
+    * @throws UnsupportedOperationException if this instance was already built
+    */
+  public Builder setXdmSchema(final Map<String, Object> xdm) {...}
+/**
+    * Builds and returns a new instance of {@code ExperienceEvent}.
+    *
+    * @return a new instance of {@code ExperienceEvent} or null if one of the required parameters is missing
+    * @throws UnsupportedOperationException if this instance was already built
+    */
+  public ExperienceEvent build() {...}
+}
+```
+
+**Examples**
+
+```java
+//Example 1
+// set freeform data to the Experience event
+Map<String, Object> xdmData = new HashMap<>();
+xdmData.put("eventType", "SampleXDMEvent");
+xdmData.put("sample", "data");
+
+Map<String, Object> data = new HashMap<>();
+data.put("free", "form");
+data.put("data", "example");
+
+ExperienceEvent experienceEvent = new ExperienceEvent.Builder()
+  .setXdmSchema(xdmData)
+  .setData(data)
+  .build();
+```
+
+```java
+//Example 2
+// Create Experience Event from XDM Schema implementations
+public class XDMSchemaExample implements com.adobe.marketing.mobile.xdm.Schema {
+  private String eventType;
+  private String otherField;
+      ...
+
+      public String getEventType() {
+        return this.eventType;
+      }
+
+      public void setEventType(final String newValue) {
+        this.eventType = newValue;
+      }
+
+      public String getOtherField() {
+        return this.otherField;
+      }
+
+      public void setOtherField(final String newValue) {
+        this.otherField = newValue;
+      }
+      }
+
+// Create Experience Event from Schema
+XDMSchemaExample xdmData = new XDMSchemaExample();
+xdmData.setEventType("SampleXDMEvent");
+xdmData.setOtherField("OtherFieldValue");
+
+ExperienceEvent experienceEvent = new ExperienceEvent.Builder()
```

Comment:
there seem to be some code missing here to set the xdmData

Reply:
Updated

Reply:
Looks good, just remove the extra semicolon at the end of L489